targettests/Remote-Sim/observe_h2.cpp

```diff
@@ -0,0 +1,128 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+// REQUIRES: remote-sim
+
+// clang-format off
+// RUN: nvq++ %cpp_std --target remote-mqpu --remote-mqpu-auto-launch 1 %s -o %t && %t
```

Comment:
Will this command use quake synthesis?

Reply:
No, I put it in there to exercise/demonstrate that the answers don't change between runs that require synthesis and runs that don't. If desired, I could remove it and put a comment near the `assert` line requesting that developers manually run library mode in the future if the `assert` line ever needs to change, but this seemed like the safest way to guarantee that the answers were correct.